File: 1065 한수/1065.cpp
#include <iostream>
#include <string>
#include <cstring>

int counter[1000 + 1];

int main() {
    std::cout.tie(nullptr);
    std::cin.tie(nullptr);
    std::ios::sync_with_stdio(false);

    memset(counter, 0, sizeof(counter));

    int n;
    std::cin >> n;
    for (int i = 1; i <= n; ++i) {
        if (i < 100) {
            counter[i] = counter[i - 1] + 1;
        } else {
            std::string numberStr = std::to_string(i);
            int numberStrLen = numberStr.length();

            bool isOk = true;
            int diff = numberStr[1] - numberStr[0];
            for (int j = 1; j < numberStrLen; ++j) {
                if (diff != numberStr[j] - numberStr[j - 1]) {
                    isOk = false;
                    break;
                }
            }

            counter[i] = isOk ? counter[i - 1] + 1 : counter[i - 1];
        }
    }

    std::cout << counter[n] << "\n";
}

File: 1065 한수/1065.js
let fs = require('fs');
let inputs = fs.readFileSync('/dev/stdin').toString();

let n = Number(inputs);
let counter = [];
for (let i = 0; i < n + 1; ++i) {
    counter.push(0);
}

for (let i = 1; i < n + 1; ++i) {
    if (i < 100) {
        counter[i] = counter[i - 1] + 1
    } else {
        let numberStr = String(i);
        let numberStrLen = numberStr.length;
        let diff = numberStr[1].charCodeAt(0) - numberStr[0].charCodeAt(0);
        let isOk = true;
        for (let j = 1; j < numberStrLen; ++j) {
            if (numberStr[j].charCodeAt(0) - numberStr[j - 1].charCodeAt(0) !== diff) {
                isOk = false;
                break;
            }
        }

        counter[i] = isOk ? counter[i - 1] + 1 : counter[i - 1];
    }
}

console.log(counter[n]);